javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -690,7 +687,9 @@ public T get(int index) {
         if ((index < 0) || (index >= length())) {
```

Comment:
`iterator` should be optimized by getting the corresponding leaf array and iterating over that (instead of finding it every time)